File: projects/caseStudyMicroservices/demo-app-service/src/main/java/com/sourabh/demoappservice/exceptions/HotelResponseEntityExceptionHandler.java
package com.sourabh.demoappservice.exceptions;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

import java.time.LocalDateTime;

@ControllerAdvice
public class HotelResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
    @ExceptionHandler(SomethingNotFoundException.class)
    public ResponseEntity<ExceptionResponse> SomethingNotFound(SomethingNotFoundException exception) {
        ExceptionResponse exceptionResponse = new ExceptionResponse(exception.getMessage(), LocalDateTime.now());
        return new ResponseEntity(exceptionResponse, HttpStatus.NOT_FOUND);
    }
}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/exception/HotelResponseEntityExceptionHandler.kt
package com.sourabh.hotelreservation.exception

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
import java.time.LocalDateTime

@ControllerAdvice
class HotelResponseEntityExceptionHandler: ResponseEntityExceptionHandler() {
    @ExceptionHandler(SomethingNotFoundException::class)
    open fun somethingNotFound(exception: SomethingNotFoundException): ResponseEntity<Any?> {
        val exceptionResponse = exception.message?.let { ExceptionResponse(it, LocalDateTime.now()) }
        return ResponseEntity<Any?>(exceptionResponse, HttpStatus.NOT_FOUND)
    }
}
